### ISSUE TITLE
An example of a "structural name"

### ISSUE THREAD

**Issue by jbrains** [riccardoNovaglia/jbrains-pos#3]
https://github.com/riccardoNovaglia/jbrains-pos/blob/5482f8ffb79e9e228b57eb3717fc6421ffcd0d8c/src/main/scala/pos/PointOfSale.scala#L13

This is not necessarily a _problem_, but this function name looks like a "structural name" to me. (See http://blog.thecodewhisperer.com/permalink/a-model-for-improving-names for more information.) What might happen if you tried to improve this name?

**Comment by riccardoNovaglia** [comment]
I agree, I cringed as I wrote it, but couldn't think of a better version at the time. As you mentioned in the videos I thought of the idea of using a guard, but I think I'd like even more the idea that this class doesn't actually receive invalid barcodes in the first place, so if the barcode is empty my pos class is not even called. I guess it depends if displaying a message when the barcode scanned is empty is a hard requirement or something we can discuss.

**Comment by jbrains** [comment]
I address this issue later in the course. In the meantime, you can think about what might be missing that would encourage you to fix this problem, and then also what would have to happen for you to need to build the missing thing.
